packages/trader/src/Modules/Trading/Containers/trade.tsx

```diff
@@ -0,0 +1,235 @@
+    const [category, setCategory] = React.useState<string>();
+    const [subcategory, setSubcategory] = React.useState<string>();
+    const [swipe_index, setSwipeIndex] = React.useState<number | undefined>(0);
+
+    const open_market = React.useMemo(() => {
+        if (try_synthetic_indices) {
+            return { category: 'synthetics' };
+        } else if (try_open_markets && category) {
+            return { category, subcategory };
+        }
+        return null;
+    }, [try_synthetic_indices, try_open_markets, category, subcategory]);
+
+    React.useEffect(() => {
+        onMount();
+        if (!is_synthetics_available) {
+            const setMarket = async () => {
+                const markets_to_search = ['forex', 'indices', 'commodities']; // none-synthetic
+                const { category: market_cat, subcategory: market_subcat } =
+                    (await getFirstOpenMarket(markets_to_search)) ?? {};
+                if (market_cat) {
+                    setCategory(market_cat);
+                    setSubcategory(market_subcat);
+                }
+            };
+
+            setMarket();
+        }
+        return () => onUnmount();
+    }, [onMount, onUnmount, getFirstOpenMarket, is_synthetics_available]);
+
+    React.useEffect(() => {
+        if (is_mobile) {
+            setDigits([]);
+        }
+        setTrySyntheticIndices(false);
+        setTryOpenMarkets(false);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [symbol, setDigits, setTrySyntheticIndices, is_synthetics_available]);
+
+    React.useEffect(() => {
+        const selectMultipliers = async () => {
+            await setContractTypes();
+
+            onChange({ target: { name: 'contract_type', value: 'multiplier' } });
+        };
+        if (should_show_multipliers_onboarding && !is_chart_loading && (is_synthetics_available || !is_market_closed)) {
+            selectMultipliers();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [should_show_multipliers_onboarding, is_chart_loading]);
+
+    const bottomWidgets = React.useCallback(({ digits: d, tick: t }: TBottomWidgetsParams) => {
+        return <BottomWidgetsMobile digits={d} tick={t} setTick={setTick} setDigits={setDigits} />;
+    }, []);
+
+    const onChangeSwipeableIndex = (index?: number) => {
+        setMobileDigitView(index === 0);
+        setIsDigitsWidgetActive(index === 0);
+        setSwipeIndex(index);
+    };
+
+    const onTryOtherMarkets = async () => {
+        if (!is_synthetics_available) {
+            setTryOpenMarkets(true);
+            setTimeout(() => setTryOpenMarkets(false));
+        } else {
+            setTrySyntheticIndices(true);
+            setTimeout(() => setTrySyntheticIndices(false));
+        }
+    };
+
+    const topWidgets = React.useCallback(
+        () => <ChartTopWidgets open_market={open_market} open={try_synthetic_indices || try_open_markets} />,
```

Comment:
removed `{ ...params }` from callback parameters since it's always an empty object. No props are passed there from SmartCharts when `topWidgets` is executed

Reply:
`open_market={open_market}` possible to use just `open_market`?